src/irmin-pack/pack_store.ml

```diff
@@ -529,11 +535,10 @@ module Maker (Index : Pack_index.S) (K : Irmin.Hash.S with type t = Index.key) :
       Lru.clear t.lru
 
     let sync t =
+      Index.sync t.pack.index;
       let former_offset = IO.offset t.pack.block in
       let offset = IO.force_offset t.pack.block in
-      if offset > former_offset then (
-        Dict.sync t.pack.dict;
-        Index.sync t.pack.index)
```

Comment:
this was to prevent an unnecessary IO for index - which I think needs to read the offset from disk to determine if it needs to sync. But if we don't update the pack, there is no need to do so for the index, no?